### PR TITLE
get all customers complete endpoint

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/main/java/com/vinyl/controller/UserController.java [hunk 1]
```diff
@@ -117,12 +114,28 @@
...
+            @ApiResponse(code = 401, message = NOT_AUTHORIZED_MESSAGE),
+            @ApiResponse(code = 400, message = BAD_REQUEST_MESSAGE),
+            @ApiResponse(code = 200, message = "You successfully got the customers")
+
+    })
+    @RequestMapping(value = "/users", method = RequestMethod.GET)
+    public ResponseEntity<?> getAllCustomers(@ApiParam(value = "Token hash to send in the request header", required = true) @RequestHeader(value = "Authorization", required = false) String auth) {
+        String token = AuthenticationHeaderHelper.getTokenHashOrNull(auth);
+
+        GetUserListDTO customers = userService.getAllCustomers(token);
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/UserController.java]: CustomerListDTO

File: src/main/java/com/vinyl/modelDTO/GetUserDTO.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package com.vinyl.modelDTO;
+
+import io.swagger.annotations.ApiModel;
+
+@ApiModel(description = "Details about customer")
+public class GetUserDTO {
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/modelDTO/GetUserDTO.java]: CustomerDTO

File: src/main/java/com/vinyl/service/UserService.java [hunk 3]
```diff
@@ -198,8 +196,28 @@ private void updateStocks(List<PurchaseProduct> orderedProducts) {
         }
     }
 
+    public GetUserListDTO getAllCustomers(String tokenHash) {
+        validateAdminToken(tokenHash);
+
+        List<User> userList = userRepository.findAll();
+        List<GetUserDTO> customers = new ArrayList<>();
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/UserService.java]: stream api

File: src/main/java/com/vinyl/modelDTO/GetUserDTO.java (hunk 2 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/modelDTO/GetUserDTO.java]: done

File: src/main/java/com/vinyl/service/UserService.java (hunk 3 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/UserService.java]: done

File: src/main/java/com/vinyl/controller/UserController.java (hunk 1 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/UserController.java]: done

File: src/test/java/com/vinyl/GetAllCustomersTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public void testWhenOk() {
+        ResponseEntity<CustomerListDTO> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(OK);
+        Assertions.assertThat(response.getBody().getUsers().size()).isEqualTo(1);
+    }
+
+    @Override
+    public ResponseEntity<CustomerListDTO> setUpHeaderAndGetTheResponse() {
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+        ResponseEntity<CustomerListDTO> response = trt.exchange("/users", HttpMethod.GET, new HttpEntity<>(headers), CustomerListDTO.class);
```
Comment (simionescuraluca) [inline src/test/java/com/vinyl/GetAllCustomersTest.java]: getUrl

**Pull request merged**